service/src/main/java/com/ge/predix/acs/privilege/management/dao/TitanMigrationManager.java

```diff
@@ -0,0 +1,66 @@
+package com.ge.predix.acs.privilege.management.dao;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("titan")
+public final class TitanMigrationManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TitanMigrationManager.class);
+
+    @Autowired
+    @Qualifier("resourceRepositoryProxy")
+    private ResourceRepositoryProxy resourceProxy;
+
+    @Autowired
+    @Qualifier("subjectRepositoryProxy")
+    private SubjectRepositoryProxy subjectProxy;
+
+    public TitanMigrationManager(final ResourceRepositoryProxy resourceProxy,
+            final SubjectRepositoryProxy subjectProxy) {
+        this.resourceProxy = resourceProxy;
+        this.subjectProxy = subjectProxy;
+    }
+
+    public TitanMigrationManager() {
+
+    }
+
+    @PostConstruct
+    public void doMigration() {
+        // TODO: Return value should reflect status of migration?
+        // Success, failure, interrupted etc ?
+        try {
+            if (this.resourceProxy.getVersion() == 0) {
```

Comment:
* this block is checking for version on the resource graph, however migrating both resource and subject. 
* lets separate the two migrations here. Version on subject graph also needs to be updated to 1